mono/mini/mini-posix.c

```diff
@@ -1003,11 +1003,13 @@ dump_native_stacktrace (const char *signal, MonoContext *mctx)
 			dump_for_merp = mono_merp_enabled ();
 #endif
 
+#ifndef DISABLE_STRUCTURED_CRASH
+			mini_register_sigterm_handler ();
```

Comment:
This should probably just be in the `!leave` block, otherwise we install a bit early, even when `!dump_for_merp` or when `DISABLE_STRUCTURED_CRASH` is defined

Reply:
Don't we want to install it if `!dump_for_merp` when `DISABLE_STRUCTURED_CRASH` is undefined?  I thought that combination meant that we're going to dump a json file, but not upload to merp (in theory that's what CI should be doing)

Reply:
Sorry, I was misreading the `!(dump_for_merp)` block (the `leave=true` is still under a #ifdef)